includes/pages/guest_login.php

```diff
@@ -282,6 +298,11 @@ function guest_login() {
           error(_("Please enter a password."));
         }
       } else {
+        if (verify_ldap_password()) {
+          $_SESSION['ldap_user']=$_REQUEST['nick'];
+          $_SESSION['ldap_pass']=$_REQUEST['password'];
```

Comment:
Wenn ich das richtig sehe, speicherst du hier das eingegebene Passwort in der Session, um es danach in die Datenbank vom Engelsystem zu übernehmen. Das ist einmal problematisch, da Passwörter so in Cookies herumliegen und zum anderen sollte bei jedem Login das Passwort neu gegen das LDAP überprüft werden, damit dortige Passwortänderungen mitbekommen werden. In der DB vom Engelsystem könnte man einfach einen leeres PW oder eine Zeichenkette, die auf LDAP verweist speichern.

Reply:
im Cookie liegt nur die Session-id, von daher wäre es unprobematisch und wenn das Passwort in der Datenbank nicht passt hätte er im nächsten Schritt LDAP gefragt. Das Problem wäre nur, dass ein geändertes Passwort noch weiter funktioniert hätte.  Jetzt steht da einfach ldap-auth in der Datenbank, das ist wohl die schönere Variante